lib/db/leveldb.go

```diff
@@ -314,12 +318,17 @@ func ldbUpdate(db *leveldb.DB, folder, device []byte, fs []protocol.FileInfo) in
 
 	var maxLocalVer int64
 	var fk []byte
+	localDev := isLocalDevice(device)
 	for _, f := range fs {
 		name := []byte(f.Name)
 		fk = deviceKeyInto(fk[:cap(fk)], folder, device, name)
 		l.Debugf("snap.Get %p %x", snap, fk)
 		bs, err := snap.Get(fk, nil)
 		if err == leveldb.ErrNotFound {
+			if localDev {
+				size.addFile(f)
```

Comment:
Invalid checks are done in addFile/removeFile. It got tedious to do it everywhere at the point it's called.